src/sqlalchemy_cratedb/support/util.py

```diff
@@ -39,3 +45,41 @@ def refresh_dirty(session, flush_context=None):
     dirty_classes = {entity.__class__ for entity in dirty_entities}
     for class_ in dirty_classes:
         refresh_table(session, class_)
+
+
+def quote_relation_name(ident: str) -> str:
+    """
+    Quote a simple or full-qualified table/relation name, when needed.
+
+    Simple:         <table>
+    Full-qualified: <schema>.<table>
+
+    Happy path examples:
+
+        foo => foo
+        Foo => "Foo"
+        "Foo" => "Foo"
+        foo.bar => foo.bar
+        foo-bar.baz_qux => "foo-bar".baz_qux
+
+    Such input strings will not be modified:
+
+        "foo.bar" => "foo.bar"
+    """
+
+    # Heuristically consider that if a quote exists at the beginning or the end
+    # of the input string, that the relation name has been quoted already.
+    if ident.startswith('"') or ident.endswith('"'):
+        return ident
+
+    # Heuristically consider if a dot is included, that it's a full-qualified
+    # identifier like <schema>.<table>. It needs to be split, in order to apply
+    # identifier quoting properly.
+    if "." in ident:
+        parts = ident.split(".")
+        if len(parts) > 2:
```

Comment:
a fully qualified name might contain more parts. i.e. the following is valid:

```sql
 SELECT crate.doc.t01.x FROM crate.doc.t01;
+---+
| x |
+---+
+---+
```

Reply:
Oh. Thanks.

Reply:
So, let's add capabilities to accept **three** fragments?

Reply:
> Which function to apply to the first fragment?

It looks like [`quote_schema()`](https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_32/lib/sqlalchemy/sql/compiler.py#L7503-L7540) effectively just calls [`quote()`](https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_32/lib/sqlalchemy/sql/compiler.py#L7542-L7592).

Reply:
```sql
cr> create table foo.bar.baz (id INT);
SQLParseException[Unexpected catalog name: foo. Only available catalog is crate]
```

Reply:
Patch adjusted. Thanks again.